hedera-node/src/main/java/com/hedera/services/state/submerkle/SolidityFnResult.java

```diff
@@ -217,7 +272,21 @@ public static SolidityFnResult fromGrpc(final ContractFunctionResult that) {
 				that.getGasUsed(),
 				that.getLogInfoList().stream().map(SolidityLog::fromGrpc).toList(),
 				that.getCreatedContractIDsList().stream().map(EntityId::fromGrpcContractId).toList(),
-				that.hasEvmAddress() ? that.getEvmAddress().getValue().toByteArray() : MISSING_BYTES);
+				that.hasEvmAddress() ? that.getEvmAddress().getValue().toByteArray() : MISSING_BYTES,
+				that.getStateChangesList().stream().collect(Collectors.toMap(
+						csc -> Address.wrap(Bytes.wrap(asEvmAddress(csc.getContractID()))),
+						csc -> csc.getStorageChangesList().stream().collect(Collectors.toMap(
+								sc -> Bytes.wrap(sc.getSlot().toByteArray()).trimLeadingZeros(),
+								sc -> Pair.of(
+										Bytes.wrap(sc.getValueRead().toByteArray()).trimLeadingZeros(),
+										!sc.hasValueWritten() ? null :
+												Bytes.wrap(sc.getValueWritten().getValue().toByteArray()).trimLeadingZeros()),
+								(l, r) -> l,
```

Comment:
This is the `toMap` merge function for when keys collide in the stream.

What's frustrating about this and the next untested line check is that we would have to create a unit test loading invalid data into the stream.  I'm OK ignoring.